zipline/pipeline/factors/factor.py

```diff
@@ -625,9 +626,7 @@ def rank(self, method='ordinal', ascending=True, mask=NotSpecified):
         """
         return Rank(self, method=method, ascending=ascending, mask=mask)
 
-    @expect_types(
-        target=Slice, correlation_length=int, mask=(Filter, NotSpecifiedType),
-    )
+    @expect_types(correlation_length=int, mask=(Filter, NotSpecifiedType))
```

Comment:
Omitting the type check here for `target` and instead relying on the type checking done on `target` in `RollingPearson`